cpp/src/io/json/nested_json_gpu.cu

```diff
@@ -1567,6 +1570,14 @@ std::pair<std::unique_ptr<column>, std::vector<column_name_info>> json_column_to
             json_col.current_offset - json_col.valid_count};
   };
 
+  auto get_child_schema = [schema](auto child_name) -> std::optional<schema_element> {
+    if (schema.has_value() &&
+        (schema.value().child_types.find(child_name) != std::end(schema.value().child_types))) {
+      return schema.value().child_types.find(child_name)->second;
+    }
```

Comment:
```suggestion
    if (schema.has_value()) {
      auto const result = schema.value().child_types.find(child_name);
      if( result != std::end(schema.value().child_types))) {
        return result->second;
      }
    }
```
Just a suggestion so `find` is not called twice.

Reply:
C++17 fancy "If statements with initializer" 
 https://en.cppreference.com/w/cpp/language/if#If_statements_with_initializer 

```suggestion
    if (schema.has_value()) {
        if (auto const result = schema.value().child_types.find(child_name); result != std::end(schema.value().child_types))) {
      	return result->second;
      }
    }
```

Reply:
I'm not a big fan of placing assignment statements inside of if clauses. The `;` certainly helps in this case.

Reply:
Thanks for your suggestions. I went with David's suggestion. Also prefer to avoid assignments in if clauses.